ddtrace/contrib/pyramid/trace.py

```diff
@@ -8,32 +9,37 @@
 import ddtrace
 from ...ext import http, AppTypes
 
+log = logging.getLogger(__name__)
+
+DD_TWEEN_NAME = 'ddtrace.contrib.pyramid:trace_tween_factory'
 
 def trace_pyramid(config):
     config.include('ddtrace.contrib.pyramid')
 
 def includeme(config):
-    config.add_tween('ddtrace.contrib.pyramid:trace_tween_factory')
+    # Add our tween just before the default exception handler
+    config.add_tween(DD_TWEEN_NAME, over=pyramid.tweens.EXCVIEW)
     # ensure we only patch the renderer once.
     if not isinstance(pyramid.renderers.RendererHelper.render, wrapt.ObjectProxy):
         wrapt.wrap_function_wrapper('pyramid.renderers', 'RendererHelper.render', trace_render)
 
 
 def trace_render(func, instance, args, kwargs):
-    # get the tracer from the request or fall back to the global version
-    def _tracer(value, system_values, request=None):
-        if request:
-            span = getattr(request, '_datadog_span', None)
-            if span:
-                return span.tracer()
-        return ddtrace.tracer
+    # If the request is not traced, we do not trace
+    request = kwargs.pop('request', {})
+    if not request:
+        log.debug("No request passed to render, will not be traced")
+        return func(*args, **kwargs)
+    span = getattr(request, '_datadog_span', None)
```

Comment:
can we move `_datadog_span` or such kind of "constants" somewhere on another module or at the top of the file? I'd prefer having such values visibile all in one place.